workflow/executor/executor.go

```diff
@@ -780,17 +791,34 @@ func (we *WorkflowExecutor) CaptureScriptResult(ctx context.Context) error {
 	return nil
 }
 
+func (we *WorkflowExecutor) finalizeOutput(ctx context.Context) error {
+	return retryutil.OnError(wait.Backoff{
+		Duration: time.Second,
+		Factor:   2,
+		Jitter:   0.1,
+		Steps:    5,
+		Cap:      30 * time.Second,
+	}, errorsutil.IsTransientErr, func() error {
+		err := we.patchTaskResultLabels(ctx, map[string]string{
+			common.LabelKeyReportOutputsCompleted: "true",
+		})
+		if apierr.IsForbidden(err) {
+			log.WithError(err).Warn("failed to patch task set, falling back to legacy/insecure pod patch, see https://argoproj.github.io/argo-workflows/workflow-rbac/")
```

Comment:
@juliev0 In accordance with the logic in `reportResult()` of `executor.go`, I am using the annotation as a backup for "legacy/insecure pod patch". Let me know if you think this is unnecessary.

Reply:
I see. Can you add a comment here to indicate that the pod annotation is being added as a backup?